impl/src/main/java/org/wildfly/extras/a2a/server/apps/jakarta/A2AServerResource.java

```diff
@@ -218,6 +240,46 @@ public static void setStreamingIsSubscribedRunnable(Runnable streamingIsSubscrib
         A2AServerResource.streamingIsSubscribedRunnable = streamingIsSubscribedRunnable;
     }
 
+    private ServerCallContext createCallContext(HttpServletRequest request, SecurityContext securityContext) {
+
+        if (callContextFactory.isUnsatisfied()) {
+            User user;
+
+            if (securityContext.getUserPrincipal() != null) {
```

Comment:
should this be `== null` instead of `!= null` since the user is being set to `UnauthenticatedUser.INSTANCE` below